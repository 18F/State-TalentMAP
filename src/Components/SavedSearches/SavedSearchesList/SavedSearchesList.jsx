import React from 'react';
import PropTypes from 'prop-types';
import SavedSearchesListResultsCard from '../SavedSearchesListResultsCard';
// import Alert from '../../Alert';
import { SAVED_SEARCH_PARENT_OBJECT, MAPPED_PARAM_ARRAY } from '../../../Constants/PropTypes';

const SavedSearchesList = ({ savedSearches, goToSavedSearch, deleteSearch, cloneSavedSearch,
mappedParams }) => {
  const savedSearchArray = [];
  savedSearches.results.slice().forEach(savedSearch => (
    savedSearchArray.push(
      <SavedSearchesListResultsCard
        savedSearch={savedSearch}
        goToSavedSearch={goToSavedSearch}
        deleteSearch={deleteSearch}
        key={savedSearch.id}
        cloneSavedSearch={cloneSavedSearch}
        mappedParams={mappedParams}
        /* pass a parentClassName that we can use from the BorderedList component */
        parentClassName="parent-list-container list-transparent"
      />,
    )
  ));
  return (
    <div className="usa-grid-full">
      {
        <div className="usa-grid-full saved-searches-list">
          {savedSearchArray.map(s => s)}
        </div>
      }
    </div>
  );
};

SavedSearchesList.propTypes = {
  savedSearches: SAVED_SEARCH_PARENT_OBJECT.isRequired,
  goToSavedSearch: PropTypes.func.isRequired,
  deleteSearch: PropTypes.func.isRequired,
  cloneSavedSearch: PropTypes.func.isRequired,
  mappedParams: MAPPED_PARAM_ARRAY.isRequired,
};

SavedSearchesList.defaultProps = {
  mappedParams: [],
};

export default SavedSearchesList;
