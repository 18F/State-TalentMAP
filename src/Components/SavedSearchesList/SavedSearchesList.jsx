import React from 'react';
import PropTypes from 'prop-types';
import SavedSearchesCard from '../SavedSearchesCard';
import { SAVED_SEARCH_PARENT_OBJECT } from '../../Constants/PropTypes';

const SavedSearchesList = ({ savedSearches, goToSavedSearch, deleteSearch }) => {
  const searchList = savedSearches.results.slice().map(savedSearch => (
    <SavedSearchesCard
      savedSearch={savedSearch}
      goToSavedSearch={goToSavedSearch}
      deleteSearch={deleteSearch}
    />
  ));
  return (
    <div className="usa-grid-full positions-section positions-section-new">
      {searchList}
    </div>
  );
};

SavedSearchesList.propTypes = {
  savedSearches: SAVED_SEARCH_PARENT_OBJECT.isRequired,
  goToSavedSearch: PropTypes.func.isRequired,
  deleteSearch: PropTypes.func.isRequired,
};

export default SavedSearchesList;
