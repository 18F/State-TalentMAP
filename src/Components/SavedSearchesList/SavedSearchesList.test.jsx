import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import SavedSearchesList from './SavedSearchesList';
import searchObjectParent from '../../__mocks__/searchObjectParent';

describe('SavedSearchesListComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <SavedSearchesList
        savedSearches={searchObjectParent}
        goToSavedSearch={() => {}}
        deleteSavedSearch={() => {}}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = shallow(
      <SavedSearchesList
        savedSearches={searchObjectParent}
        goToSavedSearch={() => {}}
        deleteSavedSearch={() => {}}
      />,
    );
    expect(wrapper.instance().props.savedSearches.results[0].id)
      .toBe(searchObjectParent.results[0].id);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <SavedSearchesList
        savedSearches={searchObjectParent}
        goToSavedSearch={() => {}}
        deleteSavedSearch={() => {}}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
