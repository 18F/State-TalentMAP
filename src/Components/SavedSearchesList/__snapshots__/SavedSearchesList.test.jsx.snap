// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedSearchesListComponent matches snapshot 1`] = `
<div
  className="usa-grid-full positions-section positions-section-new"
>
  <SavedSearchesList
    goToSavedSearch={[Function]}
    savedSearch={
      Object {
        "id": 1,
        "name": "search",
      }
    }
  />
</div>
`;
